src/server/engine_shard.cc

```diff
@@ -633,9 +629,7 @@ void EngineShard::PollExecution(const char* context, Transaction* trans) {
 
     if ((is_self && disarmed) || continuation_trans_->DisarmInShard(sid)) {
       auto bc = continuation_trans_->GetNamespace().GetBlockingController(shard_id_);
-      if (bool keep = run(continuation_trans_, false); !keep) {
-        // if this holds, we can remove this check altogether.
-        DCHECK(continuation_trans_ == nullptr);
```

Comment:
Kinda confusing if this is the case  (cont == nullptr) cause then we don't need this if statement and we can just call run. And the assignment is also not need :thinking:

Reply:
yeah, I will fix it in the following up PRs. I did not want to change the logic.